Core/GDCore/IDE/DependenciesAnalyzer.cpp

```diff
@@ -40,63 +39,38 @@ bool DependenciesAnalyzer::Analyze() {
 
 DependenciesAnalyzer::~DependenciesAnalyzer() {}
 
-bool DependenciesAnalyzer::Analyze(const gd::EventsList& events, bool isOnTopLevel) {
+bool DependenciesAnalyzer::Analyze(const gd::EventsList& events) {
   for (unsigned int i = 0; i < events.size(); ++i) {
     const gd::LinkEvent* linkEvent = dynamic_cast<const gd::LinkEvent*>(&events[i]);
     if (linkEvent) {
-      DependenciesAnalyzer analyzer(*this);
-
       gd::String linked = linkEvent->GetTarget();
       if (project.HasExternalEventsNamed(linked)) {
         if (std::find(parentExternalEvents.begin(),
                       parentExternalEvents.end(),
-                      linked) != parentExternalEvents.end())
-          return false;  // Circular dependency!
-
-        externalEventsDependencies.insert(
-            linked);  // There is a direct dependency
-        if (!isOnTopLevel) notTopLevelExternalEventsDependencies.insert(linked);
-        analyzer.AddParentExternalEvents(linked);
-        if (!analyzer.Analyze(project.GetExternalEvents(linked).GetEvents(),
-                              isOnTopLevel))
+                      linked) != parentExternalEvents.end()) {
+          // Circular dependency!
           return false;
-
+        }
+        bool isDependencyAdded = externalEventsDependencies.insert(linked).second;

```

Comment:
Ubertnit, would have made it clearer for me (I had to read the doc of std::set `insert` to understand)
```suggestion
        bool wasDependencyJustAdded = externalEventsDependencies.insert(linked).second;
```